torch/csrc/Stream.cpp

```diff
@@ -108,6 +313,14 @@ static struct PyGetSetDef THPStream_properties[] = {
 
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,cppcoreguidelines-avoid-non-const-global-variables)
 static PyMethodDef THPStream_methods[] = {
+    {"query", THPStream_query, METH_NOARGS, nullptr},
```

Comment:
@albanD If torch.Stream has these methods, then [_StreamBase](https://github.com/pytorch/pytorch/blob/266e278ccf899459285a576442719e4a212885dd/torch/_streambase.py#L4) should be useless, which is designed to capture stream's methods on dynamo, right? Similarly, `_EventBase` is also useless.

Reply:
Yes I think the end result is
- (almost) all users use torch.Stream directly
- We can remove torch.*.Stream in most cases
- For the ones left, we can migrate them to inherit from torch.Stream and not _StreamBase anymore
- We should only have them if there are device-specific feature that we do not want in the generic impl explicitly.

Reply:
@guangyey  @albanD  I attached another PR to test replacing _StreamBase and _EventBase with torch.Stream/Event. https://github.com/pytorch/pytorch/pull/123707